[CI] RRFRankClientYamlTestSuiteIT test {yaml=rrf/800_rrf_with_text_similarity_reranker_retriever/explain using rrf retriever and text-similarity} failing

**Build Scans:**
- [elasticsearch-intake #11340 / part2](https://gradle-enterprise.elastic.co/s/anqzai5wz3t2g)
- [elasticsearch-intake #11169 / part2](https://gradle-enterprise.elastic.co/s/o545mtfhdmsxs)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:rank-rrf:yamlRestTest" --tests "org.elasticsearch.xpack.rank.rrf.RRFRankClientYamlTestSuiteIT.test {yaml=rrf/800_rrf_with_text_similarity_reranker_retriever/explain using rrf retriever and text-similarity}" -Dtests.seed=1C25E59D1585D7AA -Dtests.locale=fr-NC -Dtests.timezone=Asia/Anadyr -Druntime.java=22
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Byaml%3Drrf%2F800_rrf_with_text_similarity_reranker_retriever%2Fexplain%20using%20rrf%20retriever%20and%20text-similarity%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.rank.rrf.RRFRankClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [rrf/800_rrf_with_text_similarity_reranker_retriever:332]: 
Expected: a numeric value within <1.0E-6> of <0.6666667>
     but: <0.8333334> differed by <0.16666570000000006> more than delta <1.0E-6>
```

**Issue Reasons:**
- [8.x] 2 failures in test test {yaml=rrf/800_rrf_with_text_similarity_reranker_retriever/explain using rrf retriever and text-similarity} (0.7% fail rate in 271 executions)
- [8.x] 2 failures in step part2 (3.8% fail rate in 53 executions)
- [8.x] 2 failures in pipeline elasticsearch-intake (3.8% fail rate in 53 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -381,9 +381,6 @@ tests:
 - class: org.elasticsearch.xpack.enrich.EnrichRestIT
   method: test {p0=enrich/20_standard_index/enrich stats REST response structure}
   issue: https://github.com/elastic/elasticsearch/issues/114753
-- class: org.elasticsearch.xpack.rank.rrf.RRFRankClientYamlTestSuiteIT
-  method: test {yaml=rrf/800_rrf_with_text_similarity_reranker_retriever/explain using rrf retriever and text-similarity}
-  issue: https://github.com/elastic/elasticsearch/issues/114757
 - class: org.elasticsearch.xpack.enrich.EnrichRestIT
   method: test {p0=enrich/30_tsdb_index/enrich documents over _bulk}
   issue: https://github.com/elastic/elasticsearch/issues/114761

--- a/x-pack/plugin/rank-rrf/src/yamlRestTest/resources/rest-api-spec/test/rrf/800_rrf_with_text_similarity_reranker_retriever.yml
+++ b/x-pack/plugin/rank-rrf/src/yamlRestTest/resources/rest-api-spec/test/rrf/800_rrf_with_text_similarity_reranker_retriever.yml
@@ -125,6 +125,9 @@ setup:
                               term: {
                                 topic: "science"
                               }
+                            },
+                            "sort": {
+                              "integer": "asc"
                             }
                           }
                         },
@@ -303,6 +306,9 @@ setup:
                               term: {
                                 topic: "science"
                               }
+                            },
+                            "sort": {
+                              "integer": "asc"
                             }
                           }
                         },

